pages/basePage.ts

```diff
@@ -0,0 +1,50 @@
+import { Page } from '@playwright/test'
+
+export class BasePage {
+  constructor(public readonly page: Page) {}
+
+  async clickSave() {
+    await this.page.getByRole('button', { name: 'Save and continue' }).click()
+  }
+
+  async clickSubmit() {
+    await this.page.getByRole('button', { name: 'Submit' }).click()
+  }
+
+  async clickContinue() {
+    await this.page.getByRole('button', { name: 'Continue' }).click()
+  }
+
+  async fillField(label: string, value: string) {
+    await this.page.getByLabel(label).fill(value)
+  }
+
+  async checkRadio(label: string) {
+    await this.page.getByLabel(label, { exact: true }).check()
+  }
+
+  async checkRadioInGroup(group: string, label: string) {
+    await this.page
+      .getByRole('group', {
+        name: group,
+      })
+      .getByLabel(label, { exact: true })
+      .check()
+  }
+
+  async checkCheckboxes(labels: Array<string>) {
+    const promises = [] as Array<Promise<void>>
+
+    for (let i = 0; i < labels.length; i += 1) {
+      promises.push(this.page.getByLabel(labels[i]).dispatchEvent('click'))
+    }
```

Comment:
just curious at to whether there's a reason to use a `for` loop here? I would have guessed that more "functional" approaches e.g `forEach()` would be more idiomatic in Typescript? e.g.

```ts
labels.forEach(label => promises.push(this.page.getByLabel(label).dispatchEvent('click')))
```